ci/Dockerfile

```diff
@@ -36,3 +36,4 @@ RUN cargo install cargo-llvm-cov cargo-nextest cargo-udeps cargo-hakari cargo-so
 
 ENV RUSTC_WRAPPER=sccache
 ENV SCCACHE_BUCKET=ci-sccache-bucket
+ENV CARGO_INCREMENTAL=0
```

Comment:
Does this override incremental build for non sim build? Did we show a speedup or lack of regression there?

Reply:
We already have that set for `ci-dev` profile previously.

https://github.com/risingwavelabs/risingwave/blob/90833f3bb7f63d1e080904f4aac99756aa82fa76/Cargo.toml#L95-L97

I think generally it makes not sense to have incremental build in CI, so I added the envvar as a foolproof approach.

Reply:
> Disable incremental compilation. CI builds often are closer to from-scratch builds, as changes are typically much bigger than from a local edit-compile cycle. For from-scratch builds, incremental adds an extra dependency-tracking overhead. It also significantly increases the amount of IO and the size of ./target, which make caching less effective.

https://matklad.github.io/2021/09/04/fast-rust-builds.html

Reply:
Great. I wonder why we did not make the deterministic build non-incremental in the first place then :) 

Anw, if it is faster then we should definitely do it. Btw, do we test the CI on a release build?

Reply:
> opt-level=2

Perhaps we should be testing at the same opt-level as our release build?